File: java-package/brainflow/src/main/java/brainflow/examples/CSP.java
package brainflow.examples;

import java.io.InputStream;
import java.util.Arrays;

import org.apache.commons.lang3.SystemUtils;
import org.apache.commons.lang3.tuple.MutablePair;
import org.apache.commons.lang3.tuple.Pair;

import brainflow.DataFilter;

public class CSP
{

    public static void main (String[] args) throws Exception
    {
        int n_channels = 2;
        double[] labels =
        { 0, 1 };
        double[][][] data =
        {
                {
                        { 6, 3, 1, 5 },
                        { 3, 0, 5, 1 } },
                {
                        { 1, 5, 6, 2 },
                        { 5, 1, 2, 2 } } };

        Pair<double[][], double[]> csp_data = DataFilter.get_csp (data, labels);

        System.out.print ("filters = \n");

        for (int i = 0; i < n_channels; i++)
        {
            System.out.println (Arrays.toString (csp_data.getLeft ()[i]));
        }
        System.out.println ();
    }
}

File: csharp-package/brainflow/csp/csp.cs
using System;
using brainflow;

namespace test
{
    class Csp
    {
        static void Main (string[] args)
        {
            // for demo get the CSP filters
            int n_epochs = 2;
            int n_channels = 2;
            int n_times = 4;

            double[,,] data = new double[2, 2, 4] {{{6, 3, 1, 5}, {3, 0, 5, 1}}, {{1, 5, 6, 2}, {5, 1, 2, 2}}};
            double[] labels = new double[2] {0, 1};

            Tuple<double[,], double[]> csp = DataFilter.get_csp(data, labels);

            Console.WriteLine("filters = ");
            for (int i = 0; i < n_channels; i++)
            {
                Console.WriteLine();
                for (int j = 0; j < n_channels; j++)
                {
                    Console.Write(csp.Item1[i, j] + "  ");
                }
            }
        }
    }
}
